src/Connection/Http1Connection.php

```diff
@@ -540,9 +537,6 @@ private function determineKeepAliveTimeout(Response $response): int
         $params = Http\createFieldValueComponentMap(Http\parseFieldValueComponents($response, 'keep-alive'));
 
         $timeout = (int) ($params['timeout'] ?? $this->priorTimeout);
-        if (isset($params['timeout'])) {
-            $this->explicitTimeout = true;
-        }
```

Comment:
I think we need to keep the `explicitTimeout` flag. With the current change, all connections without timeout will close after the grace period, even if it's entirely safe to reuse them for GET requests.